### DISCUSSION TITLE
Problems setting up jdtls

### DISCUSSION THREAD

**Discussion by BitInByte** [mfussenegger/nvim-jdtls#567] [Q&A] [answered]
Hello everyone, hope you're having an amazing week so far.

I've used this plugin mixed with jdtls in the past, I've even created a YouTube video to help other people to set up this amazing plugin. I've only been since then, a Java developer on my free time and now, I wanted to start a new side project with spring and I had to set up everything again to develop this side project.

The issue is that now, I've been not able to set it up. 

I've installed jdtls and created a lua file on ftplugin with the following:
```lua
local config = {
	cmd = { "~/.local/share/nvim/mason/bin/jdtls" },
	root_dir = vim.fs.dirname(vim.fs.find({ ".gradlew", ".git", "mvnw" }, { upward = true })[1]),
}
require("jdtls").start_or_attach(config)
```

The problem is that, I get the following error: 
```
Spawning language server with cmd: `~/.local/share/nvim/mason/bin/jdtls` failed. The language server is either not installed
, missing from PATH, or not executable. 
```

If I go to that location and execute the jdtls exec manually, it just executes with any problem at all. I've even tried to download a millstone as I made last time and execute from there and got the same issue.

Any suggestion on what am I doing wrong? Thanks in advance...

**Comment by mfussenegger** [comment, marked as answer]
Change `cmd = { "~/.local/share/nvim/mason/bin/jdtls" },` to `cmd = { vim.fn.expand("~/.local/share/nvim/mason/bin/jdtls") },`

`~` isn't automatically expanded to the home directory,  so it can't find the executable

**Reply by BitInByte** [reply]
Hello @mfussenegger , it worked, thank you so much. For the first time, someone have answered a topic I've created on the discussions of a repository. Thank you so much for the help. 

I'm gonna take this chance to also update my YouTube video on how to set up this plugin to work with java soon, have received some messages there saying that the steps I've made in there are not working anymore (which is normal since I made that video a while ago) so that more people will be able to easily use this plugin as well...

One more time, thank you so much, not only for answering but also for all of the things you have done for the neovim and vim community and for always being there for us. Thank you :D
